File: Domain/Entities/EventType.cs
using Domain.Common;

namespace Domain.Entities
{
    /// <summary>
    /// Entity that represents the types available for an event.
    /// </summary>
    public class EventType : BaseAuditableEntity
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? Color { get; set; }
        public Guid CreatorId { get; set; }

        public virtual ICollection<Event> TypedEvents { get; set; } = new HashSet<Event>();
    }
}

File: client-app/src/app/models/eventType.ts

export interface EventType {
    id: string;
    creatorId?: string;
    title: string;
    description: string
    color: string;
}

export const emptyEventType = (): EventType => ({
    id: '',
    title: '',
    color: '',
    creatorId: '',
    description: '',
});